CHANGELOG.md

```diff
@@ -20,26 +20,38 @@ listed in the changelog.
 - Add notifications via configurable webhook call from `ods-finish` ([#140](https://github.com/opendevstack/ods-pipeline/issues/140))
 - Git LFS support enabled ([#420](https://github.com/opendevstack/ods-pipeline/issues/420))
 - Publish images to public registry (ghcr.io) ([#440](https://github.com/opendevstack/ods-pipeline/issues/440))
-- Allow to cache dependencies ([#147](https://github.com/opendevstack/ods-pipeline/issues/147)). See also proposal on caching ([#412](https://github.com/opendevstack/ods-pipeline/pull/412))
+- Allow to cache dependencies and support for caching go dependencies ([#147](https://github.com/opendevstack/ods-pipeline/issues/147)). See also proposal on caching ([#412](https://github.com/opendevstack/ods-pipeline/pull/412))
 - Support node production builds in docker context ([#357](https://github.com/opendevstack/ods-pipeline/issues/357))
 
 ### Changed
 
 - Use configmap `ods-sonar` to configure SonarQube edition ([#410](https://github.com/opendevstack/ods-pipeline/issues/410))
 - Prevent existing image streams from being cleaned up if they are renamed in future versions ([#366](https://github.com/opendevstack/ods-pipeline/issues/366))
-- Add `build-dir` and `copy-node-modules` parameters to TypeScript build task to make it more suitable for FE builds ([#356](https://github.com/opendevstack/ods-pipeline/issues/356))
 - Update gradle version to 7.3.3 to address log4j vulnerability and improved JDK 17 support. ([#395](https://github.com/opendevstack/ods-pipeline/issues/395))
 - Create and use one PVC per repository ([#160](https://github.com/opendevstack/ods-pipeline/issues/160))
-- Leaner NodeJS 16 Typescript image task, removed cypress and its dependencies ([#426](https://github.com/opendevstack/ods-pipeline/issues/426))
 - Update skopeo (from 1.4 to 1.5) and buildah (from 1.22 to 1.23) ([#430](https://github.com/opendevstack/ods-pipeline/issues/430))
-- Use `--ignore-scripts` when building TypeScript apps ([#434](https://github.com/opendevstack/ods-pipeline/issues/434))
 - Prune pipelines and pipeline runs ([#153](https://github.com/opendevstack/ods-pipeline/issues/153)). Note that any pipeline runs created with 0.2.0 or earlier will not be cleaned up and need to be dealt with manually.
 - Log artifact URL after upload ([#384](https://github.com/opendevstack/ods-pipeline/issues/384))
 - Remove Tekton Triggers, moving the required functionality it provided into the new ODS pipeline manager ([#438](https://github.com/opendevstack/ods-pipeline/issues/438))
 - Use UBI8 provided Python 3.9 toolset image ([#457](https://github.com/opendevstack/ods-pipeline/issues/457))
 - Change installation mode from centralized to local/namespaced ([#404](https://github.com/opendevstack/ods-pipeline/pull/404))
 - Removed logging of test reports for TypeScript and Python build tasks ([#470](https://github.com/opendevstack/ods-pipeline/issues/470))
 
+#### Typescript changes
```

Comment:
shouldn't this be moved to a documentation section on its own or as an ADR?

Reply:
Hm, not sure if ADRs are a good fit for documenting breaking changes as long as they don't affect the overall architecture of `ods-pipeline`. Might a release notes document be a good fit for those kinds of documentation?

Reply:
I think the changelog is the right place. However I do not see from the headline that this is breaking, and there could be more breaking changes. How about specifying a convention at the top of the file how breaking changes are called out?

I was thinking about something like this:
```
- [Breaking] Description of the change (link)
```

Or maybe:
```
- [Breaking TypeScript Change] Description of the change ABC (link)
- [Breaking TypeScript Change] Description of the change DEF (link)
```

Reply:
I also vote for presenting the breaking changes more prominently.
Most of the time the description of changes itself mention which task is affected so it might not be necessary to mention the task (in this case TypeScript) explicitly in the brackets.
On the other hand it might be worth to be able to see in one glimpse what is affected by those breaking changes, so clustering the changes by task makes sense to me as well.

Reply:
In addition to that, once we're post-0.x and are using semantic versioning, it would be apparent from the version number that there were breaking changes. As breaking changes would affect the major version number, we'd probably ship some sort of additional docs (like the update guides in the core ods documentation) anyways, right?

Reply:
My assumption was that anything which breaks existing consumers would be in the  `Changed` section. However it may also contain non-breaking changes.

It is not always clear to me what changes are actually breaking, for example:

- Use UBI8 provided Python 3.9 toolset image ([#457](https://github.com/opendevstack/ods-pipeline/issues/457))

The previous image used Python 3.9 as well. Likely this change will only break a few consumers or nobody at all. If we are conservative we mark most changes as breaking and perhaps even some additions, but it will not be applicable to most folks updating. 

If we add a [breaking] tag it implies we have a clear definition on when we mean by that. Given that we are on a noncommittal 0.x version,  the proposed changelog changes are a compromise to not spend too much effort while still helping users to upgrade quicker. 

This issue main intend is to do that for users with typescript build tasks coming from v0.2.0.